apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -1539,6 +1539,61 @@ static void tc_pthread_pthread_setcanceltype(void)
 }
 #endif
 
+#ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
+static void *test_mutex_consistent(void *param)
+{
+	int ret = -1;
+	ret = pthread_mutex_lock(&g_mutex);
+	return (void *)ret;
+}
+
+/**
+* @fn                   :tc_pthread_pthread_mutex_consistent
+* @brief                :This tc tests pthread_mutex_consistent()
+* @Scenario             :the pthread_mutex_consistent() function can be used to mark the state protected by the mutex referenced by mutex as consistent again
+*                        and this situation arises, If an owner of a robust mutex terminates while holding the mutex,
+*                        the mutex becomes inconsistent. In this case, the mutex does not become normally usable again until the state is marked consistent.
+*                        after marking consistent next thread can acquire the lock but will receive EOWNERDEAD.
+* @API'scovered         :pthread_mutex_consistent
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void tc_pthread_pthread_mutex_consistent(void)
+{
+	int ret;
+	void *status;
+	pthread_t thread1;
+	pthread_t thread2;
+	pthread_mutexattr_t mutex_attr;
+
+	pthread_mutexattr_init(&mutex_attr);
+	pthread_mutexattr_setrobust(&mutex_attr, PTHREAD_MUTEX_ROBUST);
+	pthread_mutex_init(&g_mutex, &mutex_attr);
+
+	/* creates the 1st thread to lock mutex */
+	ret = pthread_create(&thread1, NULL, &test_mutex_consistent, NULL);
+	TC_ASSERT_EQ("pthread_create", ret, 0);
+	pthread_join(thread1, &status);
+	TC_ASSERT_EQ_CLEANUP("pthread_mutex_consistent", (int)status, 0, pthread_mutex_unlock(&g_mutex));
+
+	/* make the mutex consistent, which became inconsistent because holding thread is dead*/
+	ret = pthread_mutex_consistent(&g_mutex);
+	TC_ASSERT_EQ_CLEANUP("pthread_mutex_consistent", ret, 0, pthread_mutex_unlock(&g_mutex));
+
+	/* create 2nd thread to lock the mutex again, which is consistent now and try to acquire the lock, it should be succeed and should return EOWNERDEAD*/
+	ret = pthread_create(&thread2, NULL, &test_mutex_consistent, NULL);
+	TC_ASSERT_EQ("pthread_create", ret, 0);
+	pthread_join(thread2, &status);
+	TC_ASSERT_EQ_CLEANUP("pthread_mutex_consistent", (int)status, EOWNERDEAD, pthread_mutex_unlock(&g_mutex));
```

Comment:
Is this right to return EOWNERDEAD? Doesn't it return EOWNERDEAD without calling `pthread_mutex_consistent`?

Reply:
Sunghan's review is correct... In this TC, we called `pthread_mutex_consistent()`, so second `pthread_mutex_lock()` will not return `EOWNERDEAD`. It can successfully acquire the mutex.
So I think we need those two testcases.
Scenario1.
1) 1st thread terminates with locking the mutex.
2) 2nd thread tries to lock the mutex, then `pthread_mutex_lock()` returns `EOWNERDEAD`.
Scenario2.
1) 1st thread terminates with locking the mutex.
2) call `pthread_mutex_consistent()`
3) 2nd thread tries to lock the mutex, and it works normally.

Reply:
it is corrected now.

- When the owner of the mutex terminates without unlocking the mutex, the mutex is unlocked. The next owner of this mutex acquires the mutex with an error return of EOWNERDEAD.
- The new owner of this mutex should make the state protected by the mutex consistent. This state might have been left inconsistent when the previous owner terminated.
- If the new owner is able to make the state consistent, call pthread_mutex_consistent() for the mutex before unlocking the mutex. This marks the mutex as consistent and subsequent calls to pthread_mutex_lock() and pthread_mutex_unlock() will behave in the normal manner.